sql/catalyst/src/main/scala/org/apache/spark/sql/catalog/v2/CatalogManager.scala

```diff
@@ -27,12 +27,16 @@ import org.apache.spark.sql.internal.SQLConf
  * A thread-safe manager for [[CatalogPlugin]]s. It tracks all the registered catalogs, and allow
  * the caller to look up a catalog by name.
  */
-class CatalogManager(conf: SQLConf) extends Logging {
+class CatalogManager(conf: SQLConf, defaultSessionCatalog: TableCatalog) extends Logging {
```

Comment:
@brkyvz I agree with you that the logic was a bit confusing, so I refine it a little bit:
1. when the `V2_SESSION_CATALOG` is not set, return the default session catalog `V2SessionCatalog`.
2. when the `V2_SESSION_CATALOG` is set, try to instantiate it, and return the default session catalog if we hit problems during instantiating.